src/vm/callingconvention.h

```diff
@@ -1415,8 +1415,11 @@ void ArgIteratorTemplate<ARGITERATOR_BASE>::ComputeReturnFlags()
             }
 #endif
 
+#ifndef UNIX_X86_ABI
             if  (size <= ENREGISTERED_RETURNTYPE_INTEGER_MAXSIZE)
```

Comment:
It a common assumption in managed code that wrapping a primitive type in a struct has no overhead. This change breaks that assumption for Linux x86.

Can we avoid changing the managed calling convention, and change this for the unmanaged calling convention only?